### PR TITLE
Add rawlink() to Runnable to comply with the greenlet interface

### LINKED ISSUE #5327: Shutting down Raiden throws a runnable related exception
## Problem Definition

Shutting down Raiden after https://github.com/raiden-network/raiden/pull/5322/files seems to throw a `Runnable` related `AttributeErrror`.

```
Your account's balance is below the estimated gas reserve of 0.001795209699899765 eth. This may lead to a loss of of funds because your account will be unable to perform on-chain transactions. Please add funds to your account as soon as possible.
^CSignal received. Shutting down ...
Traceback (most recent call last):
  File "/usr/lib64/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib64/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/lefteris/ew/raiden/raiden/__main__.py", line 17, in <module>
    main()
  File "/home/lefteris/ew/raiden/raiden/__main__.py", line 13, in main
    run(auto_envvar_prefix="RAIDEN")  # pylint: disable=no-value-for-parameter
  File "/home/lefteris/.virtualenvs/raidenpy37/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/lefteris/.virtualenvs/raidenpy37/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/lefteris/.virtualenvs/raidenpy37/lib/python3.7/site-packages/click/core.py", line 1114, in invoke
    return Command.invoke(self, ctx)
  File "/home/lefteris/.virtualenvs/raidenpy37/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/lefteris/.virtualenvs/raidenpy37/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/lefteris/.virtualenvs/raidenpy37/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/lefteris/ew/raiden/raiden/ui/cli.py", line 597, in run
    app = runner.run()
  File "/home/lefteris/ew/raiden/raiden/ui/runners.py", line 248, in run
    return self._start_services()
  File "/home/lefteris/ew/raiden/raiden/ui/runners.py", line 239, in _start_services
    raise_error=True,
  File "src/gevent/greenlet.py", line 849, in gevent._greenlet.joinall
  File "src/gevent/greenlet.py", line 862, in gevent._greenlet.joinall
  File "src/gevent/_hub_primitives.py", line 125, in gevent.__hub_primitives._WaitIterator.__iter__
AttributeError: 'APIServer' object has no attribute 'rawlink'
```

That's since we are passing a non-real `greenlet` to `gevent.joinall()` and it fails inside 
`_hub_primitives.py`:

```python
    def __iter__(self):
        # When we begin iterating, we begin the timer.
        # XXX: If iteration doesn't actually happen, we
        # could leave these links around!
        if not self._begun:
            self._begun = True

            for obj in self._objects:
                obj.rawlink(self._switch)

            if self._timeout is not None:
                self._timer = self._hub.loop.timer(self._timeout, priority=-1)
                self._timer.start(self._switch, self)
        return self
``` 

## Task

We can simply add a no-op implementation of `rawlink()` so that the `Runnable` implementation adheres to the greenlet interface.

### PR DESCRIPTION
**Pull request by LefterisJP**
## Description

Fix #5327

## PR review check list

Quality check list that cannot be automatically verified.

- Safety
    - [ ] The changes respect the necessary conditions for safety (https://raiden-network-specification.readthedocs.io/en/latest/smart_contracts.html#protocol-values-constraints)
-  Code quality
    - [ ] Error conditions are handled
    - [ ] Exceptions are propagated to the correct parent greenlet
    - [ ] Exceptions are correctly classified as recoverable or unrecoverable
- Compatibility
    - [ ] State changes are forward compatible
    - [ ] Transport messages are backwards and forward compatible
- Commits
    - [ ] Have good messages
    - [ ] Squashed unecessary commits
- If it's a bug fix:
    - Regression test for the bug
        - [ ] Properly covers the bug
        - [ ] If an integration test is used, it could not be written as a unit test
- Documentation
    - [ ] A new CLI flag was introduced, is there documentation that explains usage?
- Specs
    - [ ] If this is a protocol change, are the specs updated accordingly? If so, please link PR from the specs repo.
- Is it a user facing feature/bug fix?
    - [ ] Changelog entry has been added

### CODE REVIEW

**Comment by LefterisJP** [comment]
Also want to add a test for the Runnable class and the failure we saw in #5327. Give me a moment.

**Comment by LefterisJP** [comment]
okay test added

File: raiden/utils/runnable.py [hunk 1]
```diff
@@ -98,3 +98,8 @@ def on_success(greenlet: Greenlet) -> None:
 
     def __bool__(self) -> bool:
         return bool(self.greenlet)
+
+    def rawlink(self, callback: Callable) -> None:
```
Comment (rakanalh) [inline raiden/utils/runnable.py]: Question: When would this be called? couldn't we just replace the call to `x.greenlet.rawlink` or is this internal greenlet stuff?

Comment (LefterisJP) [reply raiden/utils/runnable.py]: Check the issue. I documented it there: https://github.com/raiden-network/raiden/issues/5327

TL;DR: It's gevent internal inside `joinall`.

Comment (rakanalh) [reply raiden/utils/runnable.py]: Cleared. Thanks
